docs/DesignDocs/images/plugin-architecture-and-cicd.svg

```diff
@@ -0,0 +1 @@
d"/><text font-family="Calibri,Calibri_MSFontService,sans-serif" font-weight="400" font-size="19" transform="translate(381.485 342)">Emulator</text><path d="M540.5 286.334C540.5 279.246 546.246 273.5 553.334 273.5L604.666 273.5C611.754 273.5 617.5 279.246 617.5 286.334L617.5 388.666C617.5 395.754 611.754 401.5 604.666 401.5L553.334 401.5C546.246 401.5 540.5 395.754 540.5 388.666Z" stroke="#2F528F" stroke-width="1.33333" stroke-miterlimit="8" fill="#D0CECE" fill-rule="evenodd"/><text font-family="Calibri,Calibri_MSFontService,sans-serif" font-weight="400" font-size="19" transform="translate(564.569 320)">HW<tspan x="-9.33337" y="23">plugin</tspan><tspan x="8.74664" y="45">A</tspan></text><path d="M631.5 286.334C631.5 279.246 637.246 273.5 644.334 273.5L695.666 273.5C702.754 273.5 708.5 279.246 708.5 286.334L708.5 388.666C708.5 395.754 702.754 401.5 695.666 401.5L644.334 401.5C637.246 401.5 631.5 395.754 631.5 388.666Z" stroke="#2F528F" stroke-width="1.33333" stroke-miterlimit="8" fill="#D0CECE" fill-rule="evenodd"/><text font-family="Calibri,Calibri_MSFontService,sans-serif" font-weight="400" font-size="19" transform="translate(655.282 320)">HW<tspan x="-9.33337" y="23">plugin</tspan><tspan x="9.07996" y="45">B</tspan></text><path d="M722.5 286.334C722.5 279.246 728.246 273.5 735.334 273.5L786.666 273.5C793.754 273.5 799.5 279.246 799.5 286.334L799.5 388.666C799.5 395.754 793.754 401.5 786.666 401.5L735.334 401.5C728.246 401.5 722.5 395.754 722.5 388.666Z" stroke="#2F528F" stroke-width="1.33333" stroke-miterlimit="8" fill="#D0CECE" fill-rule="evenodd"/><text font-family="Calibri,Calibri_MSFontService,sans-serif" font-weight="400" font-size="19" transform="translate(745.995 320)">HW<tspan x="-9.33337" y="23">plugin</tspan><tspan x="9.16663" y="45">C</tspan></text><path d="M92.5001 117.5 249.001 117.5 249.001 208.625 276.75 208.625 276.75 178.25 337.5 239 276.75 299.75 276.75 269.375 249.001 269.375 249.001 360.5 92.5001 360.5Z" stroke="#2F528F" stroke-width="1.33333" stroke-miterlimit="8" fill="#DAE3F3" fill-rule="evenodd"/><text font-family="Calibri,Calibri_MSFontService,sans-serif" font-weight="400" font-size="24" transform="translate(113.186 217)">Community<tspan x="31.58" y="29">CI/CD</tspan><tspan x="-10.3133" y="58">Infrastructure</tspan></text><path d="M967.5 340.5 1011.5 296.5 1011.5 318.5 1052.26 318.5 1052.26 252.5 1209.5 252.5 1209.5 428.5 1052.26 428.5 1052.26 362.5 1011.5 362.5 1011.5 384.5Z" stroke="#2F528F" stroke-width="1.33333" stroke-miterlimit="8" fill="#D0CECE" fill-rule="evenodd"/><text font-family="Calibri,Calibri_MSFontService,sans-serif" font-weight="400" font-size="24" transform="translate(1076.89 290)">Third Party <tspan x="25.3334" y="29">CI/CD</tspan><tspan font-style="italic" x="-8.72668" y="87">(operated by </tspan><tspan font-style="italic" x="5.41333" y="116">3</tspan><tspan font-style="italic" font-size="16" x="17.5801" y="109">rd</tspan><tspan font-style="italic" x="36.7467" y="116">party )</tspan></text></g></svg>
```

Comment:
Consider changing "HW plugin" to "TEE plugin" which maybe has a better affinity for what you intend (HW differences already have a different encapsulation under OP-TEE already for example). 

Side note: It's not clear to me that we will have a single monolithic emulator distinct from individual TEEs until we resolve some finer grain design questions around the integration of various TEE "plugins" with the binary format (and embedded metadata) and loader. Depending on the testing fidelity desired of a hardware-agnostic suite of tests, it may be that we need a suite of TEE emulators (i.e. SGX and OP-TEE emulation are distinct today), which shifts some additional responsibility to the community in this proposal.

Reply:
Agree that it's a useful clarification of terminology. I will update the diagram. Thanks!